### PR TITLE
[New Rule] UAC Bypass via privileged IFileOperation

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/415

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+creation_date = "2020/10/27"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/10/27"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies User Account Control (UAC) bypass via DLL side loading. Attackers bypass UAC to stealthily execute code with
+elevated permissions.
```
Comment (threat-punter) [inline rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml:11]:
```suggestion
Identifies attempts to bypass User Account Control (UAC) via DLL side-loading. Attackers may attempt to bypass UAC to stealthily execute code with
elevated permissions.
```

File: rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Bypass UAC via Privileged IFileOperation COM interface"
```
Comment (threat-punter) [inline rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml:17]:
```suggestion
name = "UAC Bypass Attempt via Privileged IFileOperation COM Interface"
```

File: rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+references = ["https://github.com/hfiref0x/UACME"]
+risk_score = 71
```
Comment (threat-punter) [inline rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml:19]: I think `high` severity rules should have a `risk_score` of 71.

```suggestion
risk_score = 73
```

File: rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+id = "T1088"
+name = "Bypass User Account Control"
+reference = "https://attack.mitre.org/techniques/T1088/"
+
+
+[rule.threat.tactic]
+id = "TA0004"
+name = "Privilege Escalation"
+reference = "https://attack.mitre.org/tactics/TA0004/"
+
```
Comment (threat-punter) [inline rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml:50]: Shall we also add [`Hijack Execution Flow`](https://attack.mitre.org/techniques/T1574/)?

If we want to wait until we can add sub-techniques to rules then [`Hijack Execution Flow: DLL Side-Loading`](https://attack.mitre.org/techniques/T1574/002/) will be a good candidate.

**Review by threat-punter — CHANGES_REQUESTED** [review]
👍 Looks like this will be a great addition to our endpoint rules. I left a few suggestions for your review.

File: rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml (hunk 4 above)
Comment (Samirbous) [reply rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml]: make sense, we can add that as future tuning rule (to other rules as well).

**Comment by Samirbous** [comment]
Example of Alert:

image

**Review by threat-punter — APPROVED** [review]
😎

File: rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+references = ["https://github.com/hfiref0x/UACME"]
+risk_score = 73
+rule_id = "5a14d01d-7ac8-4545-914c-b687c2cf66b3"
+severity = "high"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Privilege Escalation"]
+type = "eql"
+
+query = '''
+file where event.type : "change" and process.name : "dllhost.exe" and
+
```
Comment (bm11100) [inline rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml:26]: prob dont need all the new lines in this query, but otherwise lgtm

**Review by bm11100 — APPROVED** [review]

File: rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml (hunk 5 above)
Comment (Samirbous) [reply rules/windows/privilege_escalation_uac_bypass_dll_sideloading.toml]: removed

**Pull request merged**